lib/vdsm/storage/mpathconf.py

```diff
@@ -79,6 +80,213 @@
+# scsi:spi        Parallel SCSI
+# scsi:ssa        Serial Storage Architecture
+# scsi:sbp        Firewire
+# scsi:srp        Infiniband RDMA
+# scsi:iscsi      Internet SCSI
+# scsi:sas        Serial Attached Storage
+# scsi:adt        Automation Drive Interface
+# scsi:ata        Advanced Technology Attachment
+# scsi:unspec     multipath unable to determine scsi transport
+# ccw             Channel Command Words
+# cciss           Command and Control Interface Subsystem
+# nvme            Non-Volatile Memory Express
+# undef           multipath unable to determine protocol
+#
+# We blacklist following protocols:
+# scsi:adt  local storage protocol
+# scsi:sbp  local storage protocol
+#
+# Other protocols are not blacklisted for following reasons:
+# scsi:fcp      officially supported
+# scsi:iscsi    officially supported
+# scsi:sas      not supported, but not blacklisting it to avoid breaking
+#               existing user setup, known to be used by users
+# cciss         not supported, but not blacklisting it to avoid breaking
+#               existing user setup, known to be used by users
+# scsi:srp      not supported, but not blacklisting it to avoid breaking
+#               existing user setup
+# ccw           not supported, but not blacklisting it to avoid breaking
+#               existing user setup
+# nvme          can be used with multipath setup, protocols like NVMe-oF will
+#               come up as nvme
+# scsi:spi      superseded by scsi:fcp, not a local device, no need to
+#               blacklist it
+# scsi:ssa      superseded by scsi:fcp, not a local device, no need to
+#               blacklist it
+#
+# Devices running scsi:ata and scsi:unspec are quite common and almost always
+# cannot create multipath, so it would be good to blacklist them. But in
+# general multipath can be create on top either of these devices, so we don't
+# black list them to avoid breaking anyone's setup.
+#
+# You can make blacklist more restrictive by creating drop-in config file,
+# e.g. /etc/multipath/conf.d/local.conf, and extend blacklist
+# according to your needs, e.g.
+#
+#   blacklist {{
+#       protocol "(scsi:ata|scsi:unspec)"
+#   }}
+#
+# You can also add exceptions to blacklist (see man(5) multipath.conf for more
+# details) by adding blacklist_exceptions to your drop-in configuration file,
+# e.g.
+#
+#   blacklist_exceptions {{
+#       protocol "(scsi:spi|scsi:ssa)"
+#   }}
+#
+# We blacklist all RADOS Block Device (RBD) devices.
+# When using Ceph, multipath prevents the rbd devices from being unmapped
+# and the devices remain busy.
+
+blacklist {{
+    protocol "(scsi:adt|scsi:sbp)"
+    devnode "^(rbd)[0-9]*"
+}}
+
+# Options defined here override device specific options embedded into
+# multipathd.
+
+overrides {{
+    # NOTE: see comments in default section how to compute this value.
+    no_path_retry   {no_path_retry:d}
+}}
+
+""".format(current_tag=CURRENT_TAG, no_path_retry=_NO_PATH_RETRY)
```

Comment:
str.format() is not the write tool for this file format, we have to use confusing {{ and }} everywhere. The previous code sing %(key)s and dict was better.

Template languages usually use {{key}} or {{{key}}} to avoid this issue, since actual text is unlikely to contain this.

Reply:
That's the disadvantage of `format`, yes. Still, it is already used in the same file with the escaped curly braces and all. But here it happens a lot more, so if you don't like it, I'll change it back.

Reply:
I'll try with `string.Template`, seems the best fit for this usecase. Thanks for the suggestion!